src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
Regarding the `TINYINT` comment.
Possible 0-255 values do not seem to stop the current implementation from handling explicit `TINYINT` metadata mapping it to Java byte type (-128..127):

```java
...
// ints <= 8 bits
      case Types.BIT:
      case Types.TINYINT: {
        if (optional) {
          builder.field(fieldName, Schema.OPTIONAL_INT8_SCHEMA);
        } else {
          builder.field(fieldName, Schema.INT8_SCHEMA);
        }
        break;
      }
...
// 8 bits int
      case Types.TINYINT: {
        colValue = resultSet.getByte(col);
        break;
      }
...
```

There are not that many options that we can get the column information from in the query result.
I guess we are limited to the metadata or the column name.
We can suffix the column name with the type - and to truncate it later in the Connect schema.
Or (and I think better) - use metadata.
In case the value cannot be represented in Connect for the given precision - just specify a bigger one.
E.g. if you have `NUMBER(3,0)` that are within Connect (and Java) range for byte (int8) - either use the column as is or cast to `NUMBER(3,0)`.
If your value is `NUMBER(3,0)` but is outside the Connect (and Java) range for byte (int8) - use a data type with a bigger precision, e.g. `NUMBER(5,0)` - short (int16).

In other words - using `NUMBER(3,0)` will be a way of telling kaka-connect-jdbc to map the column to `int8`.